java/arcs/core/storage/DirectStore.kt

```diff
@@ -92,7 +92,15 @@ class DirectStore<Data : CrdtData, Op : CrdtOperation, T> /* internal */ constru
     }
 
     override fun off(callbackToken: Int) {
-        return proxyManager.unregister(callbackToken)
+        proxyManager.unregister(callbackToken)
+        if (proxyManager.isEmpty()) {
+            close()
+        }
+    }
+
+    fun close() {
```

Comment:
Will the close() cause any troubles when an update is sent from driver (onReceived() callback)?

Reply:
Good catch, I think I should change onReceived() to ignore the update even if it doesn't cause trouble.